src/main/java/io/logz/guice/jersey/JerseyServer.java

```diff
@@ -55,25 +58,28 @@ private void configureServer() {
             server.addConnector(connector);
         });
 
-        WebAppContext webAppContext = new WebAppContext();
-        webAppContext.setServer(server);
+        if (websocketConfiguration != null && "/".equals(jerseyConfiguration.getContextPath())) {
+            throw new InvalidWebSocketException(
+                    "The websocket context path cannot be the same as jersey context path. " +
+                            "Either set a context path for the rest API or disable the websocket feature. " +
+                            "Also make sure that a REST resource does not have the same path as a websocket path.");
+        }
 
-        webAppContext.addFilter(GuiceFilter.class, "/*", EnumSet.allOf(DispatcherType.class));
+        Optional<ContextHandler> webAppContext = new RestContextConfigurator(server, injectorSupplier)
```

Comment:
I don't think there is a reason for `RestContextConfigurator` to return optional, 
Let's also make configure method static: 
```java
ContextHandler restContextHandler = RestContextHandlerBuilder.build(injectorSupplier, jerseyConfiguration);
handlers.add(restContextHandler);

```